src/main/java/com/example/user_management/api/controller/UserController.java

```diff
@@ -0,0 +1,50 @@
+package com.example.user_management.api.controller;
+
+import com.example.user_management.api.converter.UserConverter;
+import com.example.user_management.entity.User;
+import com.example.user_management.api.model.request.CreateUserRequestModel;
+import com.example.user_management.api.model.response.CreateUserResponseModel;
+import com.example.user_management.service.impl.UserServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/users")
+@RequiredArgsConstructor
+public class UserController {
```

Comment:
Implement General exception handler filter to produce 500 error code in case of all unknown exceptions.
Something like one of this approached https://www.baeldung.com/exception-handling-for-rest-with-spring.